exporter/elasticsearchexporter/data_stream_router_test.go

```diff
@@ -15,53 +15,53 @@ import (
 
 type routeTestCase struct {
 	name      string
-	otel      bool
+	mode      MappingMode
 	scopeName string
 	want      elasticsearch.Index
 }
 
-func createRouteTests(dsType string) []routeTestCase {
-	renderWantRoute := func(dsType, dsDataset string, otel bool) elasticsearch.Index {
-		if otel {
-			dsDataset += ".otel"
-		}
-		return elasticsearch.NewDataStreamIndex(dsType, dsDataset, defaultDataStreamNamespace)
+var renderWantRoute = func(dsType, dsDataset string, mode MappingMode) elasticsearch.Index {
```

Comment:
```suggestion
func renderWantRoute(dsType, dsDataset string) elasticsearch.Index {
```
2 nits here:
- use func instead of var
- I think it is better to avoid copying actual handling logic into test code. Test cases should be responsible for clearly stating what's expected. i.e. in this case, otel test cases should append `.otel` to `dsDataSet`.